src/EnergyPlus/OutputProcessor.hh

```diff
@@ -1240,220 +935,140 @@ void GetVariableKeyCountandType(EnergyPlusData &state,
         this->OutputInitialized = false;
         this->ProduceReportVDD = OutputProcessor::ReportVDD::No;
         this->NumHoursInMonth = 0;
         this->NumHoursInSim = 0;
-        this->ReportList.deallocate();
-        this->NumReportList = 0;
-        this->NumExtraVars = 0;
-        this->NumOfReqVariables = 0;
-        this->NumVarMeterArrays = 0;
-        this->NumEnergyMeters = 0;
-        this->MeterValue.deallocate();
-        this->TimeStepStampReportNbr = 0;
-        this->TimeStepStampReportChr.clear();
-        this->TrackingHourlyVariables = false;
-        this->DailyStampReportNbr = 0;
-        this->DailyStampReportChr.clear();
-        this->TrackingDailyVariables = false;
-        this->MonthlyStampReportNbr = 0;
-        this->MonthlyStampReportChr.clear();
-        this->TrackingMonthlyVariables = false;
-        this->YearlyStampReportNbr = 0;
-        this->YearlyStampReportChr.clear();
-        this->TrackingYearlyVariables = false;
-        this->RunPeriodStampReportNbr = 0;
-        this->RunPeriodStampReportChr.clear();
-        this->TrackingRunPeriodVariables = false;
+        this->meterValues.clear();
+        this->freqStampReportNums = {-1, -1, -1, -1, -1, -1, -1};
+        this->freqTrackingVariables = {false};
         this->TimeStepZoneSec = 0;
         this->ErrorsLogged = false;
-        this->MaxNumSubcategories = 1;
-        this->maxNumEndUseSpaceTypes = 1;
         this->isFinalYear = false;
         this->GetOutputInputFlag = true;
-        this->minimumReportFrequency = OutputProcessor::ReportingFrequency::EachCall;
+        this->minimumReportFreq = OutputProcessor::ReportFreq::EachCall;
         this->apiVarRequests.clear();
         this->ReportNumberCounter = 0;
         this->LHourP = -1;
         this->LStartMin = -1.0;
         this->LEndMin = -1.0;
         this->GetMeterIndexFirstCall = true;
         this->InitFlag = true;
-        this->keyVarIndexes.deallocate();
-        this->curKeyVarIndexLimit = 0;
-        this->varNames.deallocate();
-        this->ivarNames.deallocate();
-        this->numVarNames = 0;
-        this->TimeValue.clear();
-        this->RVariableTypes.deallocate();
-        this->IVariableTypes.deallocate();
-        this->DDVariableTypes.deallocate();
-        this->ReqRepVars.deallocate();
-        this->VarMeterArrays.deallocate();
-        this->EnergyMeters.deallocate();
-        this->EndUseCategory.deallocate();
-        this->UniqueMeterNames.clear();
-
-        this->rDummy1TS = 0.0;
-        this->rDummy2TS = 0.0;
-        this->iDummy1TS = 0;
-        this->iDummy2TS = 0;
-        this->rDummy1 = 0.0;
-        this->rDummy2 = 0.0;
-        this->iDummy1 = 0;
-        this->iDummy2 = 0;
-        this->indexGroupKey = -1;
+
+        for (int i = 0; i < (int)OutputProcessor::TimeStepType::Num; ++i) new (&this->TimeValue[i]) OutputProcessor::TimeSteps(); 
+        
+        for (int i = 0; i < (int)this->outVars.size(); ++i) delete this->outVars[i];
```

Comment:
If you remember to do these deletes in the destrctory (and really how hard is that?) then you don't need the additional layering of `unique_ptr` or `shared_ptr`.